### ISSUE TITLE
uniform and finite MPO.from_grid: nontrivial outermost bonds

### ISSUE THREAD

**Issue by jhauschild** [tenpy/tenpy#84]
Currently, the following code raises an error:
```python
from tenpy.networks.site import SpinHalfSite
from tenpy.networks.mpo import MPO

spin = SpinHalfSite(conserve='Sz')
Id, Sz, Sp, Sm = spin.Id, spin.Sz, spin.Sp, spin.Sm
N, Jxx, Jz, hz = 8, 1., 1., 0.2

grid = [[Id,   Sp,   Sm,   Sz,   -hz*Sz    ],
        [None, None, None, None, 0.5*Jxx*Sm],
        [None, None, None, None, 0.5*Jxx*Sp],
        [None, None, None, None, Jz*Sz     ],
        [None, None, None, None, Id        ]]
H = MPO.from_grids([spin] * N, [grid] * N, bc='finite', IdL=0, IdR=4)
```

The reason is that finite MPO's demand the first and last dimension to be trivial, since this simplifies the attribution of charge values.  Would be very nice, if we can lift this restriction :)

**Comment by jhauschild** [comment]
For simple enough (nearest-neighbor) Hamiltonians, this works by fixing the code for projecting the grids onto IdL/IdR. Fixed in my personal fork, to be included in the main branch soon.

**Comment by jhauschild** [comment]
The fix was merged into master as part of #111.
